app/src/main/java/com/andrew/tasky/agenda/presentation/adapters/PhotoItemAdapter.kt

```diff
@@ -36,15 +41,21 @@ class PhotoItemAdapter(
 
     override fun onBindViewHolder(holder: PhotoItemViewHolder, position: Int) {
         holder.binding.apply {
+
             if (position != currentList.size) {
-                image.setImageURI(Uri.parse(currentList[position].uriString))
-                holder.itemView.setOnClickListener {
-                    onPhotoClick(position)
+                if (currentList[position] is EventPhoto.Local) {
+                    val item = currentList[position] as EventPhoto.Local
+                    image.setImageURI(Uri.parse(item.uri.toString()))
+                    holder.itemView.setOnClickListener {
+                        onPhotoClick(position)
+                    }
+                } else {
+                    val item = currentList[position] as EventPhoto.Remote
```

Comment:
I think I commented before that I'd use if/else if you only have 2 branches.. This one here is actually an exception where I'd prefer a when expression, since you then don't need to cast the result. 
Imagine, at some point you add a 3rd type of photos to `EventPhoto`, then this piece of code might break, while a when-expression will force you to also handle the new case.